stack.py

```diff
@@ -84,95 +92,72 @@ def parse_program(code):
     label_indexes = get_label_indexes(split_program);
     # Represents ONLY instruction indexes
     # Used to map labels and index numbers.
-    current_index = 0
     for line in split_program:
-        # TODO: Disallow `@label add 1`, `add @label 1`, etc
-        # `@label; add 1` is ok though.
         parts = line.strip().split()
         # Ignore newlines 
         if not parts or is_label(parts[0]):
             continue
-#       elif any(is_label(part) for part in parts) and 'jump' not in parts:
-#            raise ValueError("The instruction, {}, contains both a label and an instruction".format(line))
-        # Process an actual instruction
-        else:
-            current_index += 1
-            has_label = False
-            prefix = []
-            args = []
-            op = None
-            for part in parts:
-                if part in sigil_to_op:
-                    part = sigil_to_op[part]
-                # A label on the same line as an instruction is an error.
-
-                if part in prefixes:
-                    prefix.append(prefixes[part])
-                elif part in ops:
-                    if op == None:
-                        op = part
-                    else:
-                        raise ValueError("Two opcodes found in instruction: " + part + " and " + op)
-                # We convert labels to an absolute jump.
-                elif is_label(part):
-                    has_label = True
-                    if op not in ['jump', 'to']:
-                        raise ValueError("Cannot use {} with a label".format(op))
-                    if part not in label_indexes:
-                        raise ValueError("The label, {}, was not defined".format(part))
-                    op = 'to'
-                    args.append(float(label_indexes[part]))
-                # Test if part is an arguement.
-                elif is_number(part):
-                    args.append(float(part))
-                else:
-                    raise ValueError("Unknown Instruction: " + line)
-            if op == None:
-                raise ValueError("No opcode in instruction")
-            yield Instr(op, args, prefix)
+        
+        # Process the instruction
+        op = None
+        args = []
+        prefix = []
+        label = None
+        for part in parts:
+            # Check for two labels in instruction.
+            if label and is_label(part):
+                raise ValueError("Two labels on instruction {}".format(line))
+
+            if part in valid_ops:
+                op = part
+            if part in sigil_to_op:
```

Comment:
Shouldn't only one of these be valid at a time? Is `if: elif:` a more appropriate case here?